pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -1183,53 +1198,177 @@ func (c *Reconciler) updatePipelineRunStatusFromInformer(ctx context.Context, pr
 		logger.Errorf("could not list TaskRuns %#v", err)
 		return err
 	}
-	updatePipelineRunStatusFromTaskRuns(logger, pr, taskRuns)
-
 	runs, err := c.runLister.Runs(pr.Namespace).List(k8slabels.SelectorFromSet(pipelineRunLabels))
 	if err != nil {
 		logger.Errorf("could not list Runs %#v", err)
 		return err
 	}
-	updatePipelineRunStatusFromRuns(logger, pr, runs)
 
-	return nil
+	updatePipelineRunStatusFromChildObjects(ctx, logger, pr, taskRuns, runs)
+
+	return validateChildObjectsInPipelineRunStatus(ctx, pr.Status)
```

Comment:
I think this function call should be moved to `updatePipelineRunStatusFromChildObjects`

Reply:
👍